server/server.go

```diff
@@ -1851,24 +1851,68 @@ func (s *Server) autorizeTopic(next handleFunc, perm user.Permission) handleFunc
 	}
 }
 
-// maybeAuthenticate reads the "Authorization" header and will try to authenticate the user
-// if it is set.
+// maybeAuthenticate will try to authenticate the user via the header configured in the "user-header" value.
+// If this doesn't work, it will fallback to the standard "Authorization" header.
 //
 //   - If auth-file is not configured, immediately return an IP-based visitor
-//   - If the header is not set or not supported (anything non-Basic and non-Bearer),
-//     an IP-based visitor is returned
-//   - If the header is set, authenticate will be called to check the username/password (Basic auth),
-//     or the token (Bearer auth), and read the user from the database
 //
 // This function will ALWAYS return a visitor, even if an error occurs (e.g. unauthorized), so
 // that subsequent logging calls still have a visitor context.
 func (s *Server) maybeAuthenticate(r *http.Request) (*visitor, error) {
-	// Read "Authorization" header value, and exit out early if it's not set
 	ip := extractIPAddress(r, s.config.BehindProxy)
 	vip := s.visitor(ip, nil)
 	if s.userManager == nil {
 		return vip, nil
 	}
+	vip, err := s.authenticateViaUserHeader(r, vip)
+	if err != nil {
+		// Fall back to native Authentication if no user-header is configured
+		vip, err = s.authenticateViaAuthHeader(r, vip)
+	}
+	return vip, err
+}
+
+// authenticateViaUserHeader tries to authenticate the user via the header defined in the "user-header"
+// configuration value if it is set.
+//
+//   - If "user-header" is not configured, immediately return an IP-based visitor
+//   - If configured header is not set an IP-based visitor is returned
+//   - If the header is set, the database will be checked if the user is existent and not marked for deletion.
+//
+// This function will ALWAYS return a visitor, even if an error occurs (e.g. unauthorized), so
+// that subsequent logging calls still have a visitor context.
+func (s *Server) authenticateViaUserHeader(r *http.Request, vip *visitor) (*visitor, error) {
+	// Check user-header from the configuration, and exit out early if it's not set
+	if len(s.config.UserHeader) <= 0 {
+		return vip, errors.New("User-header not configured")
+	}
+	username := extractUserHeader(r, s.config.UserHeader)
+	u, err := s.userManager.User(username)
+	if err != nil || u.Deleted {
+		// Unknown users are assumed anonymous, not unauthorized/401
+		return vip, errHTTPUnauthorized
+	}
+	// Authentication with user was successful
+	return s.visitor(vip.ip, u), nil
+}
```

Comment:
This logic will not just fall back to the original implementation when the user header is not configured, but also when the user was not found or other errors occur.

I'd argue that: if the user header is configured AND the user header is set by the user, THEN `authenticateViaUserHeader` should always be called, without a fallback.